### PR TITLE
Sidecar: Allow passing initial context to openAppInSideview

### PR DESCRIPTION
**Pull request by aocenas**
This allows to pass some context information to the app from the app extensions.

Allows for initializing the app from information that is provided to the extensions point, possibly initializing queries/labels/timeranges etc.

https://github.com/user-attachments/assets/5cc3fa11-df18-43e9-a2e9-a360e4a0c92e

### CODE REVIEW

File: public/app/features/plugins/components/AppRootPage.tsx [hunk 1]
```diff
@@ -113,6 +114,7 @@ export function AppRootPage({ pluginId, pluginNavSection }: Props) {
           onNavChanged={onNavChanged}
           query={queryParams}
           path={location.pathname}
+          initialContext={initialContext}
```
Comment (aocenas) [inline public/app/features/plugins/components/AppRootPage.tsx:117]: Maybe a question here is whether to pass it as a prop here or through `PluginContextProvider`. The problem with `PluginContextProvider` though seems that plugins use it to propagate their own plugin context, masking the one that Grafana passes on.

**Review by leventebalogh — COMMENTED** [review]
Hmm, what's the purpose of the this context, could you please give a concrete example? (Is it basically to share any kind of data, like linking the time selectors in the example screen recording above)? If it's more like a general blob of contextual data for the plugin (which is passed in by the view that's displaying the plugin), then probably it shouldn't live in the sidebar service.

Sorry, answering [the comment on how to pass the context around](https://github.com/grafana/grafana/pull/94417/files#r1799079486) here, too: 

> Maybe a question here is whether to pass it as a prop here or through PluginContextProvider. The problem with PluginContextProvider though seems that plugins use it to propagate their own plugin context, masking the one that Grafana passes on.

I would rather use `PluginContextProvider` for passing it down to be honest, and I probably wouldn't change the `PluginMeta`, but introduce a new field in the context for arbitrary contextual data. I agree that it's a problem that plugins nest this provider and might override the data, but I think that's probably a bad pattern, e.g. [here in the asserts app the context is initialised with the same metainfo that it already has, so it could just be removed](https://github.com/grafana/asserts-app-plugin/blob/e6faa8fe3a5588d753308ec1d73d5a79bb137ecf/src/features/App/App.tsx#L2). Even if they are nesting this context provider, they would have access to the parent context inside their `<App>` entry point component, which is the same as what we achieve with passing it down in a prop, right?

**Comment by aocenas** [comment]
@leventebalogh 
> Hmm, what's the purpose of the this context, could you please give a concrete example?

So for example in the video this is used to pass the label selectors from the logs app to the traces app. This is done once, not sure if it makes sense to sort of keep it synchronized or make it into something both apps can write to (I think that could create potentially lot of problems so I don't want to go that way right now). It is unstructured because yeah it will depend on the extension point context and what ever app implementing the app extension decides it needs from it.

> then probably it shouldn't live in the sidebar service

I sort of agree in the context I mentioned above but not sure how else to reasonably pass it. It is part of `openAppInSideview: (context?: unknown)` because I think it makes sense for this to be decision of the app extension, what it wants to pass to the app when opening it. Could store it in some other object but I think it would look basically the same. The sidecar context isn't public so it's not like you can access it from app directly right now. So I would rather then mask it behind a public `useSidecar` context for apps that would not give access to this.

> I think we should call it _context instead - it's something that will actually hold the value of the active sidecar view's context (if it's defined).

Yes but I don't want it to change over the lifetime of the app in the sidecar so I still think the `initial` in the name makes sense, similar to how uncontrolled component uses `initialValue` instead of `value`.

**Comment by aocenas** [comment]
@leventebalogh changed it to use the pluginContext. Considering it will be used something like this:
```
const pluginContext = usePluginContext();
pluginContext.initialContext
```
seems like too much "context" :D. Maybe `initialData` would be a better name?

**Comment by sunker** [comment]
@aocenas would be useful to see how this is being used in practise. Could you point me to a PR in one of your apps where this is being used?

**Comment by aocenas** [comment]
@sunker so there is an example PR in explore traces https://github.com/grafana/explore-traces/pull/102/files#diff-ae86b1a6d1d23fd8dd0d22d7a17826fd49a829f48e7a5e0ae3e3008337a4deeaR10

If you wanted to actually see anything you would also need this on the logs app side https://github.com/grafana/explore-logs/pull/828

**Pull request closed without merging**